internal/secrets/gcp_secret_manager.go

```diff
@@ -53,7 +80,15 @@ func (sm *GCPSecretManager) GetSecretValue(ctx context.Context, name string) (st
 		return "", fmt.Errorf("failed to access secret version for %v: %w", name, err)
 	}
 	logger.Infof("loaded secret value for %v", name)
-
 	plaintext := string(result.Payload.Data)
+
+	// Cache the value.
+	sm.cacheMutex.Lock()
```

Comment:
There's a subtle race here where technically there could be multiple goroutines that exist the Rlock

it may be better to just lock write lock the whole structure - return quickly on a hit and hold it longer on an update.

Reply:
it's not an usafe race- it's just that we're not technically preventing a single refresh on expiration, could be > 1

Reply:
I really wish Go had upgradable locks 😦 . Moved to a full lock.